cmd/podman/main.go

```diff
@@ -26,198 +27,159 @@ var (
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Root, "root", "", "Path to the root directory in which data, including images, is stored")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Runroot, "runroot", "", "Path to the 'run directory' where all state information is stored")
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.Runtime, "runtime", "", "Path to the OCI-compatible binary used to run containers, default is /usr/bin/runc")
+	// -s is depracated due to conflict with -s on subcommands
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.StorageDriver, "storage-driver", "", "Select which storage driver is used to manage storage of images and containers (default is overlay)")
+	rootCmd.PersistentFlags().StringSliceVar(&MainGlobalOpts.StorageOpts, "storage-opt", []string{}, "Used to pass an option to the storage driver")
+	rootCmd.PersistentFlags().BoolVar(&MainGlobalOpts.Syslog, "syslog", false, "Output logging information to syslog as well as the console")
+
+	rootCmd.PersistentFlags().StringVar(&MainGlobalOpts.TmpDir, "tmpdir", "", "Path to the tmp directory")
 
-type flagSortedAlpha struct{ flagSorted }
-
-func (a flagSortedAlpha) Less(i, j int) bool {
-	return a.flagSorted[i].GetName() < a.flagSorted[j].GetName()
+}
+func initConfig() {
+	//	we can do more stuff in here.
 }
 
-func main() {
-	debug := false
-	cpuProfile := false
-
-	if reexec.Init() {
-		return
-	}
-
-	app := cli.NewApp()
-	app.Name = "podman"
-	app.Usage = "manage pods and images"
-	app.OnUsageError = usageErrorHandler
-	app.CommandNotFound = commandNotFoundHandler
-
-	app.Version = version.Version
-
-	app.Commands = []cli.Command{
-		containerCommand,
-		exportCommand,
-		historyCommand,
-		imageCommand,
-		imagesCommand,
-		importCommand,
-		infoCommand,
-		inspectCommand,
-		pullCommand,
-		rmiCommand,
-		systemCommand,
-		tagCommand,
-		versionCommand,
-	}
-
-	app.Commands = append(app.Commands, getAppCommands()...)
-	sort.Sort(commandSortedAlpha{app.Commands})
-
-	if varlinkCommand != nil {
-		app.Commands = append(app.Commands, *varlinkCommand)
+func before(cmd *cobra.Command, args []string) error {
+	if err := libpod.SetXdgRuntimeDir(""); err != nil {
+		logrus.Errorf(err.Error())
+		os.Exit(1)
 	}
-
-	app.Before = func(c *cli.Context) error {
-		if err := libpod.SetXdgRuntimeDir(""); err != nil {
-			logrus.Errorf(err.Error())
-			os.Exit(1)
-		}
-		args := c.Args()
-		if args.Present() && rootless.IsRootless() {
-			if _, notRequireRootless := cmdsNotRequiringRootless[args.First()]; !notRequireRootless {
-				became, ret, err := rootless.BecomeRootInUserNS()
-				if err != nil {
-					logrus.Errorf(err.Error())
-					os.Exit(1)
-				}
-				if became {
-					os.Exit(ret)
-				}
+	if rootless.IsRootless() {
+		notRequireRootless := cmdsNotRequiringRootless[cmd]
+		if !notRequireRootless || !strings.HasPrefix(cmd.Use, "help") {
```

Comment:
this should be `if !notRequireRootless && !strings.HasPrefix(cmd.Use, "help") {`

Reply:
ty

Reply:
nice spot @giuseppe !